modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java

```diff
@@ -383,16 +383,17 @@ private void sendRequestContent(final MessageContext msgContext, final EndpointR
                     }
                     out = (OutputStream) msgContext.getProperty(PassThroughConstants.BUILDER_OUTPUT_STREAM);
                     if (out != null) {
-                        overflowBlob.writeTo(out);
-                        if (pipe.isStale) {
-                            throw new IOException("Target Connection is stale..");
-                        }
                         //if HTTP MEHOD = GET we need to write down the HEADER information to the wire and need
                         //to ignore any entity enclosed methods available.
                         if (HTTPConstants.HTTP_METHOD_GET.equals(msgContext.getProperty(Constants.Configuration.HTTP_METHOD)) ||
-                                RelayUtils.isDeleteRequestWithoutPayload(msgContext)) {
+                            RelayUtils.isDeleteRequestWithoutPayload(msgContext)) {
                             pipe.setSerializationCompleteWithoutData(true);
```

Comment:
It seems the `return;` statement missing here. Please check the https://github.com/wso2-support/wso2-synapse/pull/843